### PR TITLE
Use direct download instead of S3 access

### PR DESCRIPTION
**Pull request by tumido**
Move forward with the data-collector:

- [x] Instead of reaching to S3, download the file directly
- [x] Remove dependencies for preprocesor
- [x] Serve as a base image for model specific data-collectors

Jira: [AIOPS-23](https://projects.engineering.redhat.com/browse/AIOPS-23)
Related: https://github.com/RedHatInsights/e2e-deploy/pull/53

### CODE REVIEW

**Review by parsahosseini — APPROVED** [review]
Increased commenting, summarizing the general workflow, may enrich readability and logic-flow. Basic doc-strings and a module-level description can also augment said-flow; examples (in docstrings, for example) may give value, considering the threaded nature of this capability.

File: workers.py [hunk 1]
```diff
@@ -1,49 +1,103 @@
...
+        # Fetch data
         try:
-            # Fetch data
-            s3_data = s3.fetch(filesystem, bucket, uri)
-            logger.info(
-                '%s: Downloaded records %s',
-                thread.name,
-                s3_data.shape
-            )
+            resp = requests.get(source_url, stream=True)
```
Comment (AparnaKarve) [inline workers.py]: How would this GET work without any authentication to the Data source?

Comment (tumido) [reply workers.py]: Why do you think it wouldn't work? Please look up messages on Kafka and upload service docs.

Comment (tumido) [reply workers.py]: As far as I know, the credentials are provided within the url. At least the messages which comes from the upload service do that:

```json
{
    "id": null,
    "url": "https://insights-upload-perm-test.s3.amazonaws.com/8c0c420de3ab4a92b832b0f1031e5908?AWSAccessKeyId=<REDACTED>&Signature=<REDACTED>&Expires=1544701539 ",
    "service": null,
    "payload_id": "8c0c420de3ab4a92b832b0f1031e5908",
    "account": null,
    "principal": null,
    "b64_identity": null,
    "rh_account": null,
    "rh_principal": null
}
```

I've redacted the `AWSAccessKeyId` and `Signature` from the message, but the format is the same. It's designed for easy access.

Comment (AparnaKarve) [reply workers.py]: Yes, understood what you mean. Thanks.

But what if the download fails for whatever reason? 
I guess this is a similar case that we discussed in the publisher PR - failure to Upload/implement Retry
Something like that would be expected here as well, right?

Comment (tumido) [reply workers.py]: I agree, a retry should be added. I've implemented the retry logic for the POST later on, but not here. Oh, boy. I'll fix that. Thanks!

Comment (AparnaKarve) [reply workers.py]: Feel free to add the Retry for GET in a follow-up PR

**Review by AparnaKarve — APPROVED** [review]
Last commit that does the Retry for GET and POST looks good!

Can you add `#noqa` for `tests/test_service.py` to get the PR to green?
and then merge?

Thanks!!

**Pull request merged**